modules/indexer/code/zoekt/zoekt.go

```diff
@@ -0,0 +1,412 @@
+
+	// branches := []string{repo.DefaultBranch}
+	branches := []string{"HEAD"}
+
+	err = builder.Add(
+		index.Document{
+			Name:     update.Filename,
+			Content:  charset.ToUTF8DropErrors(fileContents, charset.ConvertOpts{}),
+			Branches: branches,
+		})
+	if err != nil {
+		return fmt.Errorf("error adding document with name %s: %w", update.Filename, err)
+	}
+
+	return nil
+}
+
+// Index will save the index data
+func (b *Indexer) Index(ctx context.Context, repo *repo_model.Repository, sha string, changes *internal.RepoChanges) error {
+	builder, err := newZoektIndexBuilder(b.indexDir, repo, sha)
+	if err != nil {
+		return fmt.Errorf("error creating builder: %w", err)
+	}
+
+	if len(changes.Updates) > 0 {
+		gitBatch, err := git.NewBatch(ctx, repo.RepoPath())
+		if err != nil {
+			return err
+		}
+		defer gitBatch.Close()
+
+		for _, update := range changes.Updates {
+			err := b.addUpdate(ctx, builder, gitBatch.Writer, gitBatch.Reader, update, repo)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	for _, filename := range changes.RemovedFilenames {
+		b.addDelete(builder, filename)
+	}
+
+	return builder.Finish()
+}
+
+// Delete entries by repoId
+func (b *Indexer) Delete(ctx context.Context, repoID int64) error {
+	repo, err := repo_model.GetRepositoryByID(ctx, repoID)
+	if err != nil {
+		return err
+	}
+
+	repoPathPrefix := repo.OwnerName + "%2F" + repo.Name
+
+	// remove all {repoId}_v{N}.00000.zoekt or {repoId}_v{N}.00000.zoekt.meta or {repoId}_v{N}.00000.zoekt.*.tmp in b.indexDir
+	pattern := repoPathPrefix + "_v*.00000.zoekt*"
+	matches, err := filepath.Glob(filepath.Join(b.indexDir, pattern))
+	if err != nil {
+		return fmt.Errorf("finding files to delete: %w", err)
+	}
+
+	for _, filePath := range matches {
+		if err := os.Remove(filePath); err != nil {
+			log.Error("failed to delete %s: %v", filePath, err)
+		}
+	}
+
+	tmpPattern := repoPathPrefix + "_v*.tmp"
+	tmpMatches, err := filepath.Glob(filepath.Join(b.indexDir, tmpPattern))
+	if err != nil {
+		return fmt.Errorf("finding temp files to delete: %w", err)
+	}
+
+	for _, filePath := range tmpMatches {
+		if err := os.Remove(filePath); err != nil {
+			log.Error("failed to delete temp file %s: %v", filePath, err)
+		}
+	}
+
+	return nil
+}
+
+func TransToZoektContentQueryString(s string) string {
+	return fmt.Sprintf("content:\"%s\"", s)
+}
+
+// generateZoektQuery creates a Zoekt query object based on search options
+func (b *Indexer) generateZoektQuery(_ context.Context, opts *internal.SearchOptions) (query.Q, error) {
+	keyword := opts.Keyword
+
+	// Build base content query according to search mode
+	var contentQuery query.Q
+	var err error
+
+	switch opts.SearchMode {
+	case indexer.SearchModeRegexp:
+		// Regular expression search mode
+		contentQuery, err = query.Parse(TransToZoektContentQueryString(keyword))
+		if err != nil {
+			return nil, fmt.Errorf("parse regexp keyword %q: %w", keyword, err)
+		}
+
+	case indexer.SearchModeWords:
```

Comment:
`ZoektSearchModes` doesn't support `SearchModeWords` or `SearchModeExact`, why there are case branches for them?

Reply:
`ZoektSearchModes` contains  `SearchModesExactWords()`, which contains `SearchModeWords` and `SearchModeWords`, 

I think this might be a misunderstanding.

```go
func ZoektSearchModes() []SearchMode {
	return append(SearchModesExactWords(), []SearchMode{
		{
			ModeValue:    SearchModeRegexp,
			TooltipTrKey: "search.regexp_tooltip",
			TitleTrKey:   "search.regexp",
		},
		{
			ModeValue:    SearchModeZoekt,
			TooltipTrKey: "search.zoekt_tooltip",
			TitleTrKey:   "search.zoekt",
		},
	}...)
}
```